Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Module.cs

```diff
@@ -65,6 +65,15 @@ public void Initialize(IServiceCollection serviceCollection)
             serviceCollection.AddTransient<ProductPriceDocumentChangesProvider>();
             serviceCollection.AddTransient<ProductPriceDocumentBuilder>();
             serviceCollection.AddSingleton<LogChangesChangedEventHandler>();
+
+            serviceCollection.AddScoped<PriceExportPagedDataSource>(); // Adding as scoped, because of used services (UserManager, PrincipalFactory) scoped too
+            serviceCollection.AddSingleton<Func<ExportDataQuery, PriceExportPagedDataSource>>(provider => (exportDataQuery) => CreateExportPagedDataSource<PriceExportPagedDataSource>(provider, exportDataQuery));
```

Comment:
I think `CreateExportPagedDataSource` could be removed, it could be just
```
provider => (exportDataQuery) => provider.CreateScope().ServiceProvider.GetRequiredService<PriceExportPagedDataSource>() { DataQuery = exportDataQuery})
```